### PR TITLE
Add support for retrieving the `Info.plist` from an `ObjcProvider`

### PR DESCRIPTION
**Pull request by cgrindel**
Related to #99.

- Updated the `command_line` example to include an `Info.plist`.
- Added `link_opts.bzl` API to parse linker options and retrieve information from the options.
- Added `info_plists.bzl` API to encapsulate the logic of retrieving an `Info.plist` from a target.
- Updated `target.bzl` to use `info_plists.bzl` to retrieve an `Info.plist` if present.

### CODE REVIEW

File: test/internal/info_plists/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,6 @@
+load(":get_file_from_objc_provider_tests.bzl", "get_file_from_objc_provider_test_suite")
+load(":get_file_tests.bzl", "get_file_test_suite")
+
+get_file_from_objc_provider_test_suite()
```
Comment (brentleyjones) [inline test/internal/info_plists/BUILD.bazel:4]: Can we pass the name of the suites here like in the other tests?

Comment (cgrindel) [reply test/internal/info_plists/BUILD.bazel:4]: Sure. Curious. What is the benefit of doing that way?

Comment (brentleyjones) [reply test/internal/info_plists/BUILD.bazel]: Mainly following this: https://github.com/bazelbuild/rules_apple/commit/e3ebb8f4b7b8ed8b155b49d434bc7e13e41ba092

Comment (cgrindel) [reply test/internal/info_plists/BUILD.bazel]: Interesting. I am surprised that Buildifier did not yell at me. Thanks.

**Review by brentleyjones — APPROVED** [review]
I really like the encapsulation here. I need to do that more elsewhere in the code.

FYI, I applied some small formatting changes to the branch instead of leaving them as suggested changes here.

**Pull request merged**